File: 2018-04-04/Services/app/src/main/java/br/ufpe/cin/if710/services/MainThreadServiceActivity.java
package br.ufpe.cin.if710.services;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainThreadServiceActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_thread_service);

        final Intent serviceIntent = new Intent(this,MainThreadService.class);

        final Button startButton = findViewById(R.id.btn_StartServiceMainThread);
        final Button stopButton = findViewById(R.id.btn_StopServiceMainThread);
        final Button toastButton = findViewById(R.id.btn_Toast);
        startButton.setOnClickListener(new View.OnClickListener() {
            public void onClick(View src) {
                startService(serviceIntent );
                Toast.makeText(getApplicationContext(),"Iniciou o Service",Toast.LENGTH_LONG).show();
            }
        });

        stopButton.setOnClickListener(new View.OnClickListener() {
            public void onClick(View src) {
                stopService(serviceIntent );
            }
        });
        toastButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(getApplicationContext(),"Toast",Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: 2018.1/2018-04-04/Services/app/src/main/java/br/ufpe/cin/if710/services/MainThreadServiceActivity.kt
package br.ufpe.cin.if710.services

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast

class MainThreadServiceActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_thread_service)

        val serviceIntent = Intent(this, MainThreadService::class.java)

        val startButton = findViewById<Button>(R.id.btn_StartServiceMainThread)
        val stopButton = findViewById<Button>(R.id.btn_StopServiceMainThread)
        val toastButton = findViewById<Button>(R.id.btn_Toast)
        startButton.setOnClickListener {
            startService(serviceIntent)
            Toast.makeText(applicationContext, "Iniciou o Service", Toast.LENGTH_LONG).show()
        }

        stopButton.setOnClickListener { stopService(serviceIntent) }
        toastButton.setOnClickListener { Toast.makeText(applicationContext, "Toast", Toast.LENGTH_SHORT).show() }
    }
}
